### PR TITLE
Begin arranging for cron tests

### PR DESCRIPTION
**Pull request by imurchie**
Changes to the tests to accommodate using Sauce Labs
1. Stages instead of a build matrix
1. Pull driver destruction into a helper method

### CODE REVIEW

File: test/functional/helpers/session.js [hunk 1]
```diff
@@ -4,8 +4,30 @@ import logger from '../../../lib/logger';
 import wd from 'wd';
 
 
-async function initDriver (caps, adbPort) {
-  if (process.env.TRAVIS) {
+const {SAUCE_RDC, SAUCE_EMUSIM, CLOUD, TRAVIS, CI} = process.env;
+
+function getPort () {
+  if (SAUCE_EMUSIM || SAUCE_RDC) {
```
Comment (mykola-mokhnach) [inline test/functional/helpers/session.js]: a ternary operator would also look good here

File: test/functional/helpers/session.js [hunk 2]
```diff
@@ -35,4 +66,10 @@ async function initDriver (caps, adbPort) {
   return driver;
 }
 
-export { initDriver };
+async function deleteSession () {
+  try {
+    await driver.quit();
```
Comment (mykola-mokhnach) [inline test/functional/helpers/session.js]: should we check if `driver` has value?

File: test/functional/helpers/session.js [hunk 3]
```diff
@@ -14,13 +36,22 @@ async function initDriver (caps, adbPort) {
...
-  let driver = await wd.promiseChainRemote(DEFAULT_HOST, DEFAULT_PORT);
+  const host = getHost();
+  const port = getPort();
+  logger.debug(`Starting session on ${host}:${port}`);
+  driver = await wd.promiseChainRemote(host, port);
   await driver.init(caps);
 
   // In Travis, there is sometimes a popup
-  if (process.env.CI) {
+  if (CI && !CLOUD) {
```
Comment (mykola-mokhnach) [inline test/functional/helpers/session.js]: is CI and TRAVIS the same thing? I don't really get the logic behind these conditions

File: test/functional/helpers/session.js (hunk 2 above)
Comment (imurchie) [reply test/functional/helpers/session.js]: Doesn't matter since we just ignore the error.

File: test/functional/helpers/session.js (hunk 3 above)
Comment (imurchie) [reply test/functional/helpers/session.js]: Travis sets a number of variables by default. https://docs.travis-ci.com/user/environment-variables/#default-environment-variables

In this case, Travis emulators often get a popup that blocks automation. But when using Sauce, Travis is just running the test, not the emulator. So we need to check for it only on the condition where the emulator is running within Travis' infrastructure.

**Pull request merged**